src/installer/tests/nosdk/TestApps.cs

```diff
@@ -0,0 +1,348 @@
+            {
+                WriteBasicRuntimeConfig(
+                    Path.Combine(tempDir, $"{AppName}.runtimeconfig.json"),
+                    s_provider.Tfm,
+                    s_provider.MicrosoftNETCoreAppVersion,
+                    selfContained);
+
+                DependencyContext dependencyContext = new DependencyContext(
+                    new Microsoft.Extensions.DependencyModel.TargetInfo($".NETCoreApp,Version=v{s_provider.Tfm[3..]}", null, null, true),
+                    Microsoft.Extensions.DependencyModel.CompilationOptions.Default,
+                    Enumerable.Empty<CompilationLibrary>(),
+                    new[]
+                    {
+                        new RuntimeLibrary(
+                            RuntimeLibraryType.project.ToString(),
+                            AppName,
+                            "1.0.0",
+                            string.Empty,
+                            new []
+                            {
+                                new RuntimeAssetGroup(null, $"{AppName}.dll")
+                            },
+                            Array.Empty<RuntimeAssetGroup>(),
+                            Enumerable.Empty<ResourceAssembly>(),
+                            Enumerable.Empty<Dependency>(),
+                            false)
+                    },
+                    Enumerable.Empty<RuntimeFallbacks>());
+
+                DependencyContextWriter writer = new DependencyContextWriter();
+                using (FileStream stream = new FileStream(Path.Combine(tempDir, $"{AppName}.deps.json"), FileMode.Create))
+                {
+                    writer.Write(dependencyContext, stream);
+                };
+            }
+            else
+            {
+                WriteBasicSelfContainedDepsJson(Path.Combine(tempDir, $"{AppName}.deps.json"));
+            }
+
+            HostWriter.CreateAppHost(
+                GetAppHostPath(AppName, selfContained),
+                Path.Combine(tempDir, RuntimeInformationExtensions.GetExeFileNameForCurrentPlatform(AppName)),
+                $"{AppName}.dll",
+                windowsGraphicalUserInterface: false,
+                assemblyToCopyResourcesFrom: null,
+                enableMacOSCodeSign: true);
+
+            CopySentenceSubDir(tempDir);
+        }
+
+        private static string GetAppHostPath(
+            string appName,
+            bool selfContained)
+        {
+            return selfContained
+                ? Path.Combine(s_provider.CoreClrPath, "corehost", RuntimeInformationExtensions.GetExeFileNameForCurrentPlatform("singlefilehost"))
+                : Path.Combine(s_provider.HostArtifacts, RuntimeInformationExtensions.GetExeFileNameForCurrentPlatform("apphost"));
+        }
+
+        private static void WriteBasicRuntimeConfig(
+            string outputPath,
+            string tfm,
+            string version,
+            bool selfContained)
+        {
+            var frameworkText = @$"
```

Comment:
You could use the `RuntimeConfig` class from `TestUtils`, to make this a bit shorter and avoid the verbatim JSON in code.

Reply:
Agreed. I think the equivalent for the self-contained `deps.json` below too.